dcmdump/src/main.rs

```diff
@@ -267,18 +275,17 @@ where
     I: IntoIterator,
     I::Item: std::fmt::Display,
 {
-    let pieces = values.into_iter().take(64);
-    let max = max_characters as usize;
-    let mut o = String::with_capacity(max);
-    for piece in pieces {
-        o.push_str(&piece.to_string());
-        o.push('\\');
-        if o.len() > max {
-            break;
-        }
-    }
-    o.pop();
-    cut_str(&o, max_characters).into_owned()
+    let pieces = values
+        .into_iter()
+        .take(64)
+        .map(|piece| format!("\"{}\"", piece))
+        .collect::<Vec<String>>();
```

Comment:
I also believe that you do not need to `collect` the string parts into a vector if you use [`Itertools::join`](https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.join) to join the strings.

Reply:
Although, we need to know how many items there are in order to know whether to wrap in a square brackets.